react-time-track/src/utils/calculateRisk.js

```diff
@@ -1,55 +1,48 @@
 /**
+ * Function resumeWeekly(weeklyData)
+ * Reduces estimated and real cost of the input data and returns it as an array.
+ * @param {array} weeklyData array of objects with all weekly data
+ */
+
+function resumeWeekly(weeklyData) {
+  const projectsCosts = weeklyData.reduce((accum, data) => {
+    if (data === 0) return void 0;
+    if (accum.length === 0) {
+      accum.push(data.estimated_cost);
+      accum.push(data.real_cost);
+      return accum;
+    }
+    accum[0] = accum[0] + data.estimated_cost;
+    accum[1] = accum[1] + data.real_cost;
+    return accum;
+  }, []);
+  return projectsCosts;
+}
+
+/**
+ * Function calculateRisk(weekly)
  *Sets the color whenever the project is above, according or below estimated costs.
- @param {String} start_date - string with start date for project
- @param {String} end_date - string with final date for project
- @param {String} estimated_cost - string with estimated_cost for project
- @param {String} real_cost - string with real_cost for project
+ @param {array} weekly - array of objects with all the weekly data: id, estimated_cost, real_cost, week
+ 
  */
 
-export default function calculateRisk(
-  start_date,
-  end_date,
-  estimated_cost,
-  real_cost
-) {
+function calculateRisk(weekly) {
   const red = { borderColor: "#f24636" };
   const ambar = { borderColor: "#fec235" };
   const green = { borderColor: "#52af50" };
 
   let riskValue = [];
+  let weekData = [];
+
+  if (weekly.length === 0) weekData = [0, 0];
```

Comment:
Could you please explain a bit about this code?

Reply:
Yes. The API `api/projects` returns an array of objects. Inside, there's an object that contains an array: all the weekly_project_data of the requested projects (be it the list of all the projects or an specific one). This array is sent to this function `calculateRisk` to return the appropriate color depending on comparing `real_cost` and `estimated_cost`.

But, the array *weekly* might contain several elements and also none (recently created project), as `weekly_project_reports` accumulates estimated_cost and real_costs per project per calendar week. So `calculateRisk` calls function `resumeWeekly` to return an array of two elements: the accumulated estimated_cost and the accumulated real_cost, which are compared in `calculateRisk`, which returns the color required to express the status of the project: 
red: the weekly accumulated real_cost is greater (over 100%) than the weekly accumulated estimated_cost,
yellow: the weekly accumulated real_cost is equal to or at least 90% of the weekly accumulated estimated_cost,
green: the weekly accumulated real_cost is less than 90% of the weekly accumulated estimated_cost.

But here you have found a bug. I cannot set the `weekData` array to [0, 0] (in case weekly_project_report is empty by an newly created project). This is a division by 0 error.
Thanks for checking that.